djhtml/__main__.py

```diff
@@ -51,12 +57,31 @@ def main():
             _error(e)
             continue
 
+        # Guess tabwidth
+        if not options.tabwidth:
+            diff = 0
+            probabilities = [0] * 9
+            for line in source.split("\n"):
+                count = 0
+                for char in line:
+                    if char == " ":
+                        count += 1
+                    elif char == "\t":
+                        count += 4
+                    else:
+                        break
+                if count and abs(count - diff) in [2, 4, 8]:
```

Comment:
Maybe one should only consider positive indents?

Reply:
Then we'd throw away half the information!